mempool/clist_mempool.go

```diff
@@ -176,6 +176,15 @@ func (mem *CListMempool) Unlock() {
 	mem.updateMtx.Unlock()
 }
 
+// Safe for concurrent use by multiple goroutines.
+func (mem *CListMempool) PreUpdate() {
+	rechecking := mem.recheck.done()
+	reCheckFull := mem.recheck.recheckFull.Swap(rechecking)
+	if rechecking != reCheckFull {
+		mem.logger.Info("the state of recheckFull has flipped", "before", reCheckFull, "after", rechecking)
```

Comment:
I think this is enough as a debug message. Is it necessary to print it as info?

Reply:
If anything, an info could be printed when reCheckFull == true to inform the validator that their mempool has problematic transactions that slow down transaction checks.